tests/PathTest.php

```diff
@@ -822,4 +822,80 @@ public function testIsBasePath($path, $ofPath, $result)
     {
         $this->assertSame($result, Path::isBasePath($path, $ofPath));
     }
+
+    public function provideCombineTests()
+    {
+        $object = new ObjectConvertibleToString('object/path');
+
+        return array(
+            array(null, null, null),
+            array('/path/to/test', null, '/path/to/test'),
+            array('/path/to//test', null, '/path/to/test'),
+            array(null, '/path/to/test', '/path/to/test'),
+            array(null, '/path/to//test', '/path/to/test'),
+
+            array('/path/to/test', 'subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', 'subdir', '/path/to/test/subdir'),
+            array('/path/to/test', '/subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', '/subdir', '/path/to/test/subdir'),
+            array('/path/to/test', './subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', './subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', '../parentdir', '/path/to/parentdir'),
+            array('/path/to/test', '../parentdir', '/path/to/parentdir'),
+            array('path/to/test/', '/subdir', 'path/to/test/subdir'),
+            array('path/to/test', '/subdir', 'path/to/test/subdir'),
+            array('../path/to/test', '/subdir', '../path/to/test/subdir'),
+            array('path', '../../subdir', '../subdir'),
+            array('/path', '../../subdir', '/subdir'),
+            array('../path', '../../subdir', '../../subdir'),
+
+            array('/path/to/test', 123, '/path/to/test/123'),
+            array('/path/to/test', $object, '/path/to/test/object/path'),
+
+            array(array('/path/to/test', 'subdir'), null, '/path/to/test/subdir'),
+            array(array('/path/to/test', '/subdir'), null, '/path/to/test/subdir'),
+            array(array('/path/to/test/', 'subdir'), null, '/path/to/test/subdir'),
+            array(array('/path/to/test/', '/subdir'), null, '/path/to/test/subdir'),
+
+            array(array('/path'), null, '/path'),
+            array(array('/path', 'to', '/test'), null, '/path/to/test'),
+            array(array('/path', null, '/test'), null, '/path/test'),
+            array(array('path', 'to', 'test'), null, 'path/to/test'),
+
+            array('C:\\path\\to\\test', 'subdir', 'C:/path/to/test/subdir'),
+            array('C:\\path\\to\\test\\', 'subdir', 'C:/path/to/test/subdir'),
+            array('C:\\path\\to\\test', '/subdir', 'C:/path/to/test/subdir'),
+            array('C:\\path\\to\\test\\', '/subdir', 'C:/path/to/test/subdir'),
```

Comment:
1.2 adds support for schemes (e.g. `phar://C:/foo/bar`, `phar:///foo/bar`, ...). Could you rebase this PR and add corresponding test cases?